scripts/zones/Southern_San_dOria/npcs/Kipopo.lua

```diff
@@ -6,18 +6,41 @@
 -----------------------------------
 require("scripts/globals/status");
 require("scripts/globals/crafting");
+require("scripts/globals/quests");
 local ID = require("scripts/zones/Southern_San_dOria/IDs");
 -----------------------------------
 
 function onTrade(player,npc,trade)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
+
+    if (player:hasKeyItem(dsp.ki.TORN_PATCHES_OF_LEATHER) and sayItWithAHandbagCS == 2 and trade:getItemCount() == 3) then
+        if (trade:hasItemQty(2012,1) and trade:hasItemQty(850,1) and trade:hasItemQty(816,1)) then
+        player:startEvent(910)
+        end
+    end
 end;
 
 function onTrigger(player,npc)
+    local sayItWithAHandbag = player:getQuestStatus(CRYSTAL_WAR, dsp.quest.id.crystalWar.SAY_IT_WITH_A_HANDBAG)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
     local guildMember = isGuildMember(player,7);
     local SkillCap = getCraftSkillCap(player, dsp.skill.LEATHERCRAFT);
     local SkillLevel = player:getSkillLevel(dsp.skill.LEATHERCRAFT);
 
-    if (guildMember == 1) then
+    if (sayItWithAHandbag == QUEST_COMPLETED and sayItWithAHandbagCS == 4 and player:hasItem(19110)) then
+        player:startEvent(914)
+        player:setVar("sayItWithAHandbagCS", 5)
```

Comment:
What's this 5 being used for?

Reply:
@ibm2431  The used is so that CS is only played once, as Kipopo is supposed to go back to her standard dialog ```player:startEvent(651) -- Standard Dialogue``` once quest is done as ```player:startEvent(914)``` is an optional CS not part of the questline.

Reply:
another var handles Bonus CS. No longer using 5 anymore. Thank you for your feedback 😸